test/common/config/version_converter_test.cc

```diff
@@ -16,7 +16,7 @@ namespace {
 // Wire-style upgrading between versions.
 TEST(VersionConverterTest, Upgrade) {
   API_NO_BOOST(envoy::api::v2::Cluster) source;
-  source.set_drain_connections_on_host_removal(true);
+  API_NO_BOOST(source.set_drain_connections_on_host_removal(true));
```

Comment:
Ideally I think you can infer the type of `source` in the clang-tools so this line can be skipped without macro, no?

Reply:
Yeah, the type can be inferred, but not sure if I can get retrieve the fact that the original type declaration was wrapped in a macro, which is erased. I think it might be possible in simple cases, where the object expression is a simple variable to chain back via https://clang.llvm.org/doxygen/classclang_1_1CXXMemberCallExpr.html#a4dfa9b9a8cf5feda21a268378c3a283d, will look into this.

Reply:
Yeah I don't expect that can handle all case, but some simple case like this can be cleaner. This is just nice-to-have.

Reply:
@lizan updated to do inference on best-effort basis, PTAL.